### PR TITLE
Added Quotas

### PR DESCRIPTION
**Pull request by 69mb**
Added the following configurable limits:
- Number of drops per river in an account
- Number of rivers an account can have
- Number of channel options  per type
- Lifetime of the river(Updated river expiry)

### CODE REVIEW

File: application/classes/controller/river/maintenance.php [hunk 1]
```diff
@@ -76,20 +66,18 @@ public function action_run()
...
 			$river_url = $site_url.$river->get_base_url();
 
-			// Generate extension token and modify the URL
 			if ($days_to_expiry === 0)
 			{
-				$token = hash("sha256", Text::random('alnum', 32));
-				$river_url .= '/extend?token='.$token;
-				$to_be_expired[$river->id] = $token;
+				$to_be_expired[] = $river->id;
+				Swiftriver_Event::run('swiftriver.river.disable', $river);
```
Comment (ekala) [inline application/classes/controller/river/maintenance.php:75]: Document this plugin hook - `swiftriver.river.disable`\- here: https://wiki.ushahidi.com/display/WIKI/Plugin+Hooks

File: install/sql/swiftriver.sql [hunk 2]
```diff
@@ -843,7 +846,10 @@ INSERT INTO `settings` (`id`, `key`, `value`) VALUES
 (5, 'anonymous_access_enabled', '0'),
 (6, 'river_active_duration', '14'),
 (7, 'river_expiry_notice_period', '3'),
-(8, 'general_invites_enabled', '0');
+(8, 'general_invites_enabled', '0'),
+(9, 'default_river_quota', '1'),
+(10, 'default_river_drop_quota', '10000'),
+(11, 'site_url', 'http://www.example.com');
```
Comment (ekala) [inline install/sql/swiftriver.sql:852]: The `river_active_duration` setting has been renamed to `default_river_lifetime`?

**Comment by ekala** [comment]
The license header, `@package` and `@license` doc tags need to read as follows:

```
 PHP version 5
 LICENSE: This source file is subject to the AGPL license 
 that is available through the world-wide-web at the following URI:
 http://www.gnu.org/licenses/agpl.html
 ...
 @package    SwiftRiver - https://github.com/ushahidi/SwiftRiver
 ...
 ...
 @license    http://www.gnu.org/licenses/agpl.html GNU Affero General Public License (AGPL)`
```

The affected files are:
- application/classes/controller/settings/quotas.php
- application/classes/model/account/channel/quota.php
- application/classes/model/channel/quota.php

File: application/classes/model/droplet.php [hunk 3]
```diff
@@ -429,15 +440,54 @@ public static function add_metadata(& $drops)
...
+			{
+				if (($river_quota['drop_count'] >= $river_quota['drop_quota']) OR ! (bool)$river_quota['river_active'])
+				{
+					// River already over quota or disabled
+					$over_quota_rivers[] = $river_quota['id'];
+				}
+				else if ($river_quota['drop_count'] + $rivers[$river_quota['id']] >= $river_quota['drop_quota'])
+				{
+					// Signal interested parties that this river's quota has been reached
+					Swiftriver_Event::run('swiftriver.river.quota_reached', $river_quota['id']);
```
Comment (ekala) [inline application/classes/model/droplet.php:462]: Add the plugin hook to this list - https://wiki.ushahidi.com/display/WIKI/Plugin+Hooks

**Comment by 69mb** [comment]
Fixed the license and missing setting.

**Comment by ekala** [comment]
+1

**Pull request merged**